### ISSUE TITLE
`anzu-query-replace-regexp` when use previous regex text contains '\,()'

### ISSUE THREAD

**Issue by Whiothes** [emacsorphanage/anzu#97]
#### Question:
I can replace text contains \,(func ) for the first time, 
but I can not use it again or othes  based on it. (type <↑> and use it again)

#### regexp text
```
\(\w+\)\s-+\([a-z_]+\).*?///< \(.*?\)$ -> {"\2", &app_data->mer_st.\2, "\3", PTS_DB_TYPE_\,(upcase \1), sizeof(app_data->mer_st.\2)},
```
#### Debug Info
```
Debugger entered--Lisp error: (error "Invalid use of ‘\\’ in replacement text")
  replace-match("{\"\\2\", &app_data->mer_st.\\2, \"\\3\", PTS_DB_TYPE_\\,(..." nil nil #("bigint  mer_map_id;                     ///< 商户id" 0 19 (fontified t) 19 40 (fontified t c-is-sws t c-in-sws t) 40 41 (face font-lock-comment-delimiter-face fontified t c-is-sws t c-in-sws t) 41 43 (face font-lock-comment-delimiter-face fontified t c-in-sws t) 43 49 (face font-lock-comment-face fontified t c-in-sws t)) nil)
  match-substitute-replacement("{\"\\2\", &app_data->mer_st.\\2, \"\\3\", PTS_DB_TYPE_\\,(..." nil nil)
  perform-replace("\\(\\w+\\)\\s-+\\([a-z_]+\\).*?///< \\(.*?\\)$" "{\"\\2\", &app_data->mer_st.\\2, \"\\3\", PTS_DB_TYPE_\\,(..." t t nil nil nil 5137 6729)
  apply(perform-replace ("\\(\\w+\\)\\s-+\\([a-z_]+\\).*?///< \\(.*?\\)$" "{\"\\2\", &app_data->mer_st.\\2, \"\\3\", PTS_DB_TYPE_\\,(..." t t nil nil nil 5137 6729))
  anzu--query-replace-common(t :prefix-arg 1)
  anzu-query-replace-regexp(1)
  funcall-interactively(anzu-query-replace-regexp 1)
  call-interactively(anzu-query-replace-regexp nil nil)
  command-execute(anzu-query-replace-regexp)
```

#### Steps
1. call anzu-query-replace-regexp
2. type ↑ to  find a previous regex text with function
3. modify or not , then press ENTER

Is there any more details I need to offer here?

Thanks a lot !

**Comment by syohex** [comment]
Sorry late reply. This is a bug. And I have sent PR #101

**Comment by gonewest818** [comment]
Closed by #101.
